scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/utils/TaskKiller.java

```diff
@@ -67,11 +69,26 @@ private void interruptRepeatedly() {
         }
     }
 
+    /**
+     * Interrupts and gives the thread time to cleanup
+     */
+    private void interruptAndWaitCleanupTime() {
+        threadToKill.interrupt();
+        int iterationsToHitWaitingTimeLimit = CLEANUP_SECONDS_AFTER_FIRST_INTERRUPT * 100;
+        for(int i = 0; threadToKill.isAlive() && i < iterationsToHitWaitingTimeLimit ; i++ ) {
+            try {
```

Comment:
Extract to private method these 4 duplicated lines of code? or at least use constant for sleep time?